client/components/happychat/test/composer.jsx

```diff
@@ -22,61 +25,74 @@ describe( '<Composer />', () => {
 					translate={ noop }
 				/>
 			);
-			wrapper.find( 'textarea' ).simulate( 'change', { target: { value: 'hey' } } );
+			const el = screen.getByRole( 'textbox', { name: /support request/i } );
+			await user.type( el, 'hey' );
 			expect( onSetCurrentMessage ).toHaveBeenCalled();
 			expect( onSendTyping ).toHaveBeenCalled();
 			expect( onSendNotTyping ).not.toHaveBeenCalled();
 		} );
 
-		test( 'should call onSetCurrentMessage property and send a noTyping event if message is empty', () => {
+		test( 'should call onSetCurrentMessage property and send a noTyping event if message is empty', async () => {
+			const user = userEvent.setup();
 			const onSendNotTyping = jest.fn();
 			const onSendTyping = jest.fn();
 			const onSetCurrentMessage = jest.fn();
-			const wrapper = mount(
-				<Composer
-					message={ '' }
+			const WrappedWithMessageState = ( props ) => {
+				const [ message, setMessage ] = useState( 'x' );
+				props.onSetCurrentMessage.mockImplementation( ( value ) => setMessage( value ) );
+				return <Composer { ...props } message={ message } />;
+			};
+			render(
+				<WrappedWithMessageState
 					onSetCurrentMessage={ onSetCurrentMessage }
 					onSendTyping={ onSendTyping }
 					onSendNotTyping={ onSendNotTyping }
 					translate={ noop }
 				/>
 			);
-			wrapper.find( 'textarea' ).simulate( 'change', { target: { value: '' } } );
+			const el = screen.getByRole( 'textbox', { name: /support request/i } );
+			await user.click( el );
+			await user.keyboard( '{Backspace}' );
```

Comment:
Or we could use the `clear` helper: https://testing-library.com/docs/user-event/utility#clear

Then we won't need the wrapped component at all I think?

Reply:
Yup, changed that in https://github.com/Automattic/wp-calypso/pull/66599/commits/c04631408c7552a4aa9a989ba173f52a9c105ad8